### PR TITLE
chore: rename artifact files in windows vhd pipelines

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
rename artifact files in windows vhd pipelines.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/list-images.ps1 [hunk 1]
```diff
@@ -10,7 +10,7 @@ $windowsImageVersion = $env:WindowsImageVersion
 
 $ErrorActionPreference = "Stop"
 
-$imageBomJsonFilePath = "c:\image-bom.json"
+$imageBomJsonFilePath = "c:\image-list.json"
```
Comment (AbelHu) [inline vhdbuilder/packer/list-images.ps1:13]: This file will exist inside every VHD. I think that we should not change this file name to keep it to be consistent with Linux. We only need to rename it before uploading it to artifacts.

Comment (junjiezhang1997) [reply vhdbuilder/packer/list-images.ps1]: Thanks for the comment. Just find it and the empty file of image-list.json in artifacts should be caused by this change.

File: .pipelines/templates/.builder-release-template-windows.yaml [hunk 2]
```diff
@@ -130,23 +130,38 @@ steps:
...
+  - bash: |
+      if [[ ${{ parameters.artifactName }} =~ "2019" ]]; then
+        WINDOWS_VERSION="$(cat vhdbuilder/packer/windows-image.env | grep -a "WINDOWS_2019_BASE_IMAGE_VERSION" | cut -d "=" -f 2 | cut -d "." -f 1,2)"
+      elif [[ ${{ parameters.artifactName }} =~ "2022" ]]; then
+        WINDOWS_VERSION="$(cat vhdbuilder/packer/windows-image.env | grep -a "WINDOWS_2022_BASE_IMAGE_VERSION" | cut -d "=" -f 2 | cut -d "." -f 1,2)"
+      else
+        echo "Current distro is not supported to get image version. You need to update related code."
+        exit 1
+      fi
+      IMAGE_VERSION="$WINDOWS_VERSION.${BUILD_DATE}"
```
Comment (AbelHu) [inline .pipelines/templates/.builder-release-template-windows.yaml:143]: It seems like that `${BUILD_DATE}` is empty
image

Comment (junjiezhang1997) [reply .pipelines/templates/.builder-release-template-windows.yaml:143]: Modify the code for obtaining BUILD_DATE.

File: .pipelines/templates/.builder-release-template-windows.yaml [hunk 3]
```diff
@@ -50,7 +50,7 @@ steps:
       if [[ "${USE_RELEASE_DATE}" = "False" ]]; then \
         echo "use current date as build date";  BUILD_DATE=$(date +"%y%m%d"); \
       else \
-        echo "use release date as build date"; branchName=$(Build.SourceBranchName); BUILD_DATE=${branchName: -6}; \
+        echo "use release date as build date"; BUILD_DATE=${RELEASE_DATE}; \
```
Comment (AbelHu) [inline .pipelines/templates/.builder-release-template-windows.yaml:53]: It is better to validate whether `${RELEASE_DATE}` is valid, for example, not null, not empty, the format (`yymmdd`).

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**